aws/internal/service/cognitoidentityprovider/waiter/waiter.go

```diff
@@ -0,0 +1,58 @@
+package waiter
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	// Maximum amount of time to wait for an Operation to return Success
+	UserPoolDomainDeleteTimeout = 1 * time.Minute
+	UserPoolDomainCreateTimeout = 1 * time.Minute
+)
+
+// UserPoolDomainDeleted waits for an Operation to return Success
+func UserPoolDomainDeleted(conn *cognitoidentityprovider.CognitoIdentityProvider, domain string) (*cognitoidentityprovider.DescribeUserPoolDomainOutput, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			cognitoidentityprovider.DomainStatusTypeUpdating,
+			cognitoidentityprovider.DomainStatusTypeDeleting,
+		},
+		Target:  []string{""},
```

Comment:
I think the "correct" way to do this is `Target: []string{},`:

https://github.com/terraform-providers/terraform-provider-aws/blob/691d0645ceed9453a8fc77a88e7fb0fce29b1ad3/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource/state.go#L122-L131

Reply:
this failed for me 
```
        Error: error waiting for User Pool Domain (tf-acc-test-domain-3112558782247577448) deletion: unexpected state '', wanted target ''. last error: %!s(<nil>)
```

Reply:
Now worried about how other code works 😄 ...

Reply:
Yes, that's what's happening here:

```
2020/08/19 10:14:59 [DEBUG] [aws-sdk-go] DEBUG: Response cognito-idp/DescribeUserPoolDomain Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Content-Length: 24
Content-Type: application/x-amz-json-1.1
Date: Wed, 19 Aug 2020 14:14:59 GMT
X-Amzn-Requestid: abe5941d-5d1e-4153-a926-84cfbfce7f91


-----------------------------------------------------
2020/08/19 10:14:59 [DEBUG] [aws-sdk-go] {"DomainDescription":{}}
```